openraft/src/engine/engine_impl.rs

```diff
@@ -746,6 +766,9 @@ impl<NID: NodeId> Engine<NID> {
     }
 
     // --- Draft API ---
+    pub fn update_snapshot_last_log(&mut self, id: Option<LogId<NID>>) {
+        self.snapshot_last_log_id = id;
+    }
```

Comment:
My suggestion is to remove this method.
The logic of installing a snapshot will, in the future, be moved into `Engine` thus from outside `Engine`; there won't be necessary to update `Engine.snapshot_last_log_id`.

By the way, make Engine methods `pub(crate)` or private, since the Engine is not stable yet and will probably change in the near future.